Userland/Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -641,32 +641,8 @@ JS::Interpreter& Document::interpreter()
             // FIXME: This isn't exactly the right place for this.
             HTML::main_thread_event_loop().perform_a_microtask_checkpoint();
 
-            // Note: This is not an exception check for the promise jobs, they will just leave any
-            // exception that already exists intact and never throw a new one (without cleaning it
-            // up, that is). Taking care of any previous unhandled exception just happens to be the
-            // very last thing we want to do, even after running promise jobs.
-            if (auto* exception = vm.exception()) {
-                auto value = exception->value();
-                if (value.is_object()) {
-                    auto& object = value.as_object();
-                    auto name = object.get_without_side_effects(vm.names.name).value_or(JS::js_undefined());
-                    auto message = object.get_without_side_effects(vm.names.message).value_or(JS::js_undefined());
-                    if (name.is_accessor() || message.is_accessor()) {
-                        // The result is not going to be useful, let's just print the value. This affects DOMExceptions, for example.
-                        dbgln("\033[31;1mUnhandled JavaScript exception:\033[0m {}", value);
-                    } else {
-                        dbgln("\033[31;1mUnhandled JavaScript exception:\033[0m [{}] {}", name, message);
-                    }
-                } else {
-                    dbgln("\033[31;1mUnhandled JavaScript exception:\033[0m {}", value);
-                }
-                for (auto& traceback_frame : exception->traceback()) {
-                    auto& function_name = traceback_frame.function_name;
-                    auto& source_range = traceback_frame.source_range;
-                    dbgln("  {} at {}:{}:{}", function_name, source_range.filename, source_range.start.line, source_range.start.column);
-                }
-            }
-
+            // FIXME: Do we need to handle unhandled javascript exceptions here?
+            //        Because vm.exception() does not exist anymore we cannot just 'know' exceptions were thrown.
```

Comment:
That's unfortunate and will massively decrease the development experience when using browser. Given that promise jobs don't actually throw exceptions, we don't need to do this *here*, but we still need to do it *somewhere* - wherever `Interpreter::run()` is being called seems like the right place, given that we get a completion from that.

Reply:
Also note that this is being fully removed in #12308.

Reply:
I have moved the printing of the error to both `Document::run_javascript` and `ClassicScript::run` do we want/need it anywhere else?

Reply:
Seems fine.

Reply:
After some testing I expanded it all calls of interpreter.run and JS::call

Reply:
Fixme still needs to be removed